sklearn/metrics/_ranking.py

```diff
@@ -34,7 +34,7 @@
 from ..utils.validation import _deprecate_positional_args
 from ..exceptions import UndefinedMetricWarning
 from ..preprocessing import label_binarize
-from ..preprocessing._label import _encode
+from ..preprocessing._label import _encode, _unique
```

Comment:
should we move these out of `preprocessing._label` and put these in `utils`, since they're not just used for encoding labels but also features?